pkg/skaffold/schema/v2beta29/upgrade.go

```diff
@@ -124,6 +153,19 @@ func upgradeOnePipeline(oldPipeline, newPipeline interface{}) error {
 		newPL.Deploy.LegacyHelmDeploy = &next.LegacyHelmDeploy{}
 		newPL.Deploy.LegacyHelmDeploy.LifecycleHooks = newHelm.LifecycleHooks
 	}
+	return nil
+}
 
+func upgradePatches(olds []JSONPatch, news []next.JSONPatch) error {
+	for i, old := range olds {
+		for str, repStr := range migrations {
+			if strings.Contains(old.Path, str) {
+				news[i].Path = strings.Replace(old.Path, str, repStr, 0)
+			}
+			if strings.Contains(old.Path, "/deploy/kpt") {
+				fmt.Println("skip migrating kpt deploy sections. Please migrate these over manually")
```

Comment:
should this be `Println` or `log.Entry`?  Also would it make sense to error in the case `/deploy/kpt` is found in a patch profile  stating the above message (is there ever a case where the old patch profile would continue to work) vs printing it out and continuing?  It might make sense not to error, just thinking through the # of error cases vs correct cases possible

Reply:
I was initially inclined to return an error. However, I was wondering if it made sense to convert the main skaffold.yaml.  A user would probably had not configured kpt as a deployer in main skaffold.yaml. (The code would error out when upgrading the pipeline)

e.g.
```
apiVersion: skaffold/v2beta29
kind: Config
deploy:
  manifests:
    k8s:
    - 'base-service/*.yaml'
    
profiles:
   - name: try-kpt
      patches:
       - op: add
          path: /deploy/kpt/
          value:
              <>
```

This could be an edge case and maybe I am over-engineering ?

Reply:
we could still use `log.Entry.Warn` instead of `Println`